flow-html-components/src/main/java/com/vaadin/flow/component/html/RangeInput.java

```diff
@@ -0,0 +1,271 @@
+    private static final PropertyDescriptor<String, String> maxDescriptor = PropertyDescriptors
+            .attributeWithDefault("max", "100");
+    private static final PropertyDescriptor<String, String> stepDescriptor = PropertyDescriptors
+            .attributeWithDefault("step", "1");
+    private static final PropertyDescriptor<String, String> orientDescriptor = PropertyDescriptors
+            .attributeWithDefault("orient", Orientation.HORIZONTAL.getValue());
+
+    /**
+     * The orientation of the range slider.
+     */
+    public enum Orientation {
+        HORIZONTAL("horizontal"), VERTICAL("vertical");
+
+        private final String value;
+
+        Orientation(String value) {
+            this.value = value;
+        }
+
+        public String getValue() {
+            return value;
+        }
+    }
+
+    private int valueChangeTimeout = DEFAULT_CHANGE_TIMEOUT;
+
+    private ValueChangeMode currentMode;
+
+    /**
+     * Creates a new slider, with {@link ValueChangeMode#ON_CHANGE ON_CHANGE}
+     * value change mode.
+     */
+    public RangeInput() {
+        this(ValueChangeMode.ON_CHANGE);
+    }
+
+    /**
+     * Creates a new slider.
+     *
+     * @param valueChangeMode
+     *            initial value change mode, or <code>null</code> to disable the
+     *            value synchronization
+     */
+    public RangeInput(ValueChangeMode valueChangeMode) {
+        super("value", 0.0, false);
+        setValueChangeMode(valueChangeMode);
+        set(typeDescriptor, "range");
+    }
+
+    @Override
+    public ValueChangeMode getValueChangeMode() {
+        return currentMode;
+    }
+
+    @Override
+    public void setValueChangeMode(ValueChangeMode valueChangeMode) {
+        currentMode = valueChangeMode;
+        setSynchronizedEvent(
+                ValueChangeMode.eventForMode(valueChangeMode, "input"));
+        applyChangeTimeout();
+    }
+
+    @Override
+    public void setValueChangeTimeout(int valueChangeTimeout) {
+        this.valueChangeTimeout = valueChangeTimeout;
+        applyChangeTimeout();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * The default value is {@link HasValueChangeMode#DEFAULT_CHANGE_TIMEOUT}.
+     */
+    @Override
+    public int getValueChangeTimeout() {
+        return valueChangeTimeout;
+    }
+
+    private void applyChangeTimeout() {
+        ValueChangeMode.applyChangeTimeout(currentMode, valueChangeTimeout,
+                getSynchronizationRegistration());
+    }
+
+    /**
+     * Returns the minimum value.
+     *
+     * @return the minimum value, defaults to 0.
+     */
+    public double getMin() {
+        return Double.parseDouble(get(minDescriptor));
+    }
+
+    /**
+     * Sets the new minimum value.
+     *
+     * @param min
+     *            the minimum value.
+     */
+    public void setMin(double min) {
+        set(minDescriptor, "" + min);
+    }
+
+    /**
+     * Returns the maximum value.
```

Comment:
nit: Flow team conventions, use "gets" instead of "returns"
```suggestion
     * Gets the maximum value.
```

Reply:
Done.